src/pages/index.js

```diff
@@ -0,0 +1,291 @@
+const imageAvatar = document.getElementById("image-avatar");
+const imagePenBlack = document.getElementById("image-pen");
+const imagePenWhite = document.getElementById("image-pen-white");
+const imagePlus = document.getElementById("image-plus");
+
+imageLogo.src = logoSrc;
+imagePenBlack.src = penSrc;
+imagePenWhite.src = penWhiteSrc;
+imagePlus.src = plusSrc;
+
+const profileEditButton = document.querySelector(".profile__edit-btn");
+const editModal = document.querySelector("#edit-modal");
+
+const photoModal = document.querySelector("#photo-modal");
+const photoModalImage = photoModal.querySelector(".modal__photo");
+const photoModalCaption = photoModal.querySelector(".modal__description");
+
+const profileFormElement = document.forms["edit-profile"];
+const profileSaveBtn = profileFormElement.querySelector(".modal__save-button");
+
+const nameInput = profileFormElement.querySelector("#name");
+const jobInput = profileFormElement.querySelector("#description");
+
+const profileNameElement = document.querySelector(".profile__name");
+const profileJobElement = document.querySelector(".profile__description");
+
+api
+  .getAppInfo()
+  .then((result) => {
+    const savedCards = result[0];
+    const savedProfile = result[1];
+    imageAvatar.src = savedProfile["avatar"];
+    imageAvatar.alt = savedProfile["name"];
+    profileNameElement.textContent = savedProfile["name"];
+    profileJobElement.textContent = savedProfile["about"];
+    savedCards.forEach((card) => {
+      const cardEl = getCardElements(card);
+      cardsList.append(cardEl);
+    });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", escapeListener);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", escapeListener);
+}
+
+function escapeListener(evt) {
+  const modal = document.querySelector(".modal_opened");
+  if (evt.key === "Escape") {
+    closeModal(modal);
+  }
+}
+
+const modalList = document.querySelectorAll(".modal");
+modalList.forEach((modal) => {
+  modal.addEventListener("click", function (evt) {
+    if (evt.target.parentElement.classList.contains("page")) {
+      closeModal(modal);
+    }
+  });
+});
+
+const closeButtons = document.querySelectorAll(".modal__close-button");
+closeButtons.forEach((button) => {
+  const popup = button.closest(".modal");
+  button.addEventListener("click", (evt) => {
+    if (button.classList.contains("modal__close-button_cancel")) {
+      evt.preventDefault();
+    }
+    closeModal(popup);
+  });
+});
+
+profileEditButton.addEventListener("click", () => {
+  openModal(editModal);
+
+  nameInput.value = profileNameElement.textContent;
+  jobInput.value = profileJobElement.textContent;
+
+  profileSaveBtn.disabled = true;
+  profileSaveBtn.classList.add("modal__save-button_disabled");
+});
```

Comment:
You should create function `disableButton` in `validation.js` for disabling not to duplicate the code. Then you need to import it into `index.js`